### PR TITLE
[Enhancement] Support profile for only big query (2)

### PR DESCRIPTION
**Pull request by liuyehcf**
## Why I'm doing:

In the previous pr https://github.com/StarRocks/starrocks/pull/33825, we've supported the profile only for big query with the session variable `big_query_profile_second_threshold`

Unfortunately, for some of our uses, most of the queries can be done within 200ms with a latency spike over 400ms, we cannot catch the slow query(400ms) with the session variable mentioned above.

## What I'm doing & Behavior Change

* Remove session variable `big_query_profile_second_threshold` with type `int`.
* Add session variable `big_query_profile_threshold` with type `string`, it can support time duration like `1s`/`500ms`/`15m` etc.

This change won't lead to compatible problem when performing grey upgrading.
* Only upgrade some of the backends, it works fine with the previous session variable `big_query_profile_second_threshold`.
* Only upgrade all of the backends, it works fine with the previous session variable `big_query_profile_second_threshold`
* And after upgrading the frontend, the user need to set up the `big_query_profile_threshold` corresponding to the previous configuration manually.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -930,7 +931,10 @@ public static MaterializedViewRewriteMode parse(String str) {
     private boolean enableAsyncProfile = true;
 
     @VariableMgr.VarAttr(name = BIG_QUERY_PROFILE_SECOND_THRESHOLD)
-    private int bigQueryProfileSecondThreshold = 0;
+    private long bigQueryProfileSecondThreshold = 0;
+
+    @VariableMgr.VarAttr(name = BIG_QUERY_PROFILE_MILLISECOND_THRESHOLD, flag = VariableMgr.INVISIBLE)
```
Comment (allencelerdata) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:936]: 1. The `BIG_QUERY_PROFILE_MILLISECOND_THRESHOLD` can replace `BIG_QUERY_PROFILE_SECOND_THRESHOLD`?
2. If you think the unit of `BIG_QUERY_PROFILE_SECOND_THRESHOLD` is too small, maybe `BIG_QUERY_PROFILE_THRESHOLD=10m/10s/10ms`, support time unit is a better name?

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:936]: Yes, only keep `BIG_QUERY_PROFILE_MILLISECOND_THRESHOLD` or provide a unit variable is a better choice. But If we do so, there may be a compatibility problem, that's why I keep them both.

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:936]: I think it over again, you're right, offer another unit variable is a better choice and don't have compatible problem.

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -929,8 +932,11 @@ public static MaterializedViewRewriteMode parse(String str) {
     @VariableMgr.VarAttr(name = ENABLE_ASYNC_PROFILE, flag = VariableMgr.INVISIBLE)
     private boolean enableAsyncProfile = true;
 
-    @VariableMgr.VarAttr(name = BIG_QUERY_PROFILE_SECOND_THRESHOLD)
-    private int bigQueryProfileSecondThreshold = 0;
+    @VariableMgr.VarAttr(name = BIG_QUERY_PROFILE_THRESHOLD, alias = BIG_QUERY_PROFILE_SECOND_THRESHOLD)
+    private long bigQueryProfileThreshold = 0;
+
+    @VariableMgr.VarAttr(name = BIG_QUERY_PROFILE_THRESHOLD_UNIT)
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: Updated.

**Review by chaoyli — DISMISSED** [review]

File: docs/en/administration/query_profile_overview.md [hunk 3]
```diff
@@ -16,10 +16,17 @@ SET enable_profile = true;
...
+It is not recommended to enable Query Profile in a production environment on a global, long-term basis. This is because the data collection and processing of Query Profile may impose additional burdens on the system. However, if you need to capture and analyze slow queries, you can enable Query Profile only for slow queries. This can be achieved by setting the variable `big_query_profile_threshold` to a time duration greater than `0s`. For example, if this variable is set to `30s`, it means that only queries with an execution time exceeding 30 seconds will trigger Query Profile. This ensures system performance while effectively monitoring slow queries.
 
 ```SQL
-SET global big_query_profile_second_threshold = 30;
+-- 30 seconds
+SET global big_query_profile_threshold = 30s;
```
Comment (kangkaisen) [inline docs/en/administration/query_profile_overview.md:23]: what's the TTimeUnit if user set big_query_profile_threshold = 30?

File: docs/zh/administration/query_profile_overview.md [hunk 4]
```diff
@@ -16,10 +16,17 @@ SET enable_profile = true;
 
 ### 针对慢查询开启 Query Profile
 
-在生产环境中，通常不推荐全面启用 Query Profile 功能。这是因为 Query Profile 的数据采集和处理过程可能会为系统带来额外的负担。然而，如果需要捕捉到耗时的慢查询，就需要巧妙地使用这一功能。为此，您可以选择只对慢查询启用 Query Profile。这可以通过设置变量 `big_query_profile_second_threshold` 为一个大于 `0` 的整数来实现。例如，若将此变量设置为 30，意味着只有那些执行时间超过 30 秒的查询会启用 Query Profile 功能。这样既保证了系统性能，又能有效监控到慢查询。
+在生产环境中，通常不推荐全面启用 Query Profile 功能。这是因为 Query Profile 的数据采集和处理过程可能会为系统带来额外的负担。然而，如果需要捕捉到耗时的慢查询，就需要巧妙地使用这一功能。为此，您可以选择只对慢查询启用 Query Profile。这可以通过设置变量 `big_query_profile_threshold` 为一个大于 `0s` 的时间来实现。例如，若将此变量设置为 `30s`，意味着只有那些执行时间超过 30 秒的查询会启用 Query Profile 功能。这样既保证了系统性能，又能有效监控到慢查询。
```
Comment (packy92) [inline docs/zh/administration/query_profile_overview.md:20]: Now,  is the unit necessary when setting this?

**Review by packy92 — APPROVED** [review]

File: docs/zh/administration/query_profile_overview.md (hunk 4 above)
Comment (liuyehcf) [reply docs/zh/administration/query_profile_overview.md:20]: yes, the type of this variable is string now, a time unit is mandatory.

File: docs/en/administration/query_profile_overview.md (hunk 3 above)
Comment (liuyehcf) [reply docs/en/administration/query_profile_overview.md]: updated. the type is str, time unit is mandatory.

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**